### DISCUSSION TITLE
Task - Control Flow Statements - Conditional Statements - Looping Statements - while loop

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#156] [Tasks]
1. Write a program to calculate the sum of the first 10 natural numbers.

2. Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number. 

3. Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24

4. Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method)

5. Determine and print the number of times the character ‘a’ appears in the input entered by the user.

6. Write a program to print a box of #’s taking height and width values from the user.

7. You have to design the code such that the user has only three tries to guess the correct pin of the account. You set the pin as a constant with a final attribute. When correct display “Correct, welcome back.” When incorrect display “Incorrect, try again.”. When ran out of tries display “Sorry but you have been locked out.”

**Comment by SathyaGanes** [comment]
Code #
```Javascript
//1. Write a program to calculate the sum of the first 10 natural numbers

let sum = 0;
for (let i = 1; i <= 10; i++) {
  sum += i;
  if (i == 10) {
    console.log("The sum of the first 10 natural numbers is " + sum);
  }
}


//2..Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number. 

let num = prompt("Enter a positive integer:");
if (num > 0) {
  for (let i = 1; i <= 10; i++) {
    console.log(`${num} x ${i} = ${num*i}`);
  }
} else if (num == 0) {
  console.log("The multiplication table of 0 is 0");
} else {
  console.log("Invalid input. Please enter a positive integer.");
}

//3.Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24


let num = prompt("Enter a positive integer:");
let factorial = 1;

if (num >= 0) {
  for (let i = num; i > 0; i--) {
    factorial *= i;
  }
  console.log(`Factorial of ${num} is ${factorial}`);
} else {
  console.log("Invalid input. Please enter a positive integer.");
}


//4..Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method) 

let base = prompt("Enter the base number:");
let exponent = prompt("Enter the exponent number:");
let result = 1;

if (exponent >= 0) {
  for (let i = 1; i <= exponent; i++) {
    result *= base;
  }
  console.log(`${base} raised to the power of ${exponent} is ${result}`);
} else {
  console.log("Invalid input. Exponent should be a non-negative integer.");
}


//5.Determine and print the number of times the character ‘a’ appears in the input entered by the user. 
let input = prompt("Enter a string:");
let count = 0;

for (let i = 0; i < input.length; i++) {
  if (input[i] === "a") {
    count++;
  }
}

console.log(`The character 'a' appears ${count} times in the input.`);

//6.Write a program to print a box of #’s taking height and width values from the user. 

let height = prompt("Enter the height of the box:");
let width = prompt("Enter the width of the box:");

if (height > 0 && width > 0) {
  // Print top edge of the box
  let top = "";
  for (let i = 1; i <= width; i++) {
    top += "#";
  }
  console.log(top);

  // Print middle rows of the box
  for (let i = 2; i <= height - 1; i++) {
    let middle = "#";
    for (let j = 2; j <= width - 1; j++) {
      middle += " ";
    }
    middle += "#";
    console.log(middle);
  }

  // Print bottom edge of the box
  if (height > 1) {
    let bottom = "";
    for (let i = 1; i <= width; i++) {
      bottom += "#";
    }
    console.log(bottom);
  }
} else {
   console.log("Invalid input. Height and width should be positive integers.");
}

7. You have to design the code such that the user has only three tries to guess the correct pin of the account. You set the pin as a constant with a final attribute. When correct display “Correct, welcome back.” When incorrect display “Incorrect, try again.”. When ran out of tries display “Sorry but you have been locked out.”

const PIN = "1234";
let tries = 3;

while (tries > 0) {
  let input = prompt(`Enter the PIN (${tries} tries left):`);
  if (input === PIN) {
    console.log("Correct, welcome back.");
    break;
  } else {
    console.log("Incorrect, try again.");
    tries--;
  }
}

if (tries === 0) {
  console.log("Sorry but you have been locked out.");
}

```

**Comment by sam2elect** [comment]
```js
//1. Write a program to calculate the sum of the first 10 natural numbers.

let sum = 0;
for (let i = 1; i <= 10; i++) {
  sum += i;
}
console.log("The sum of the first 10 natural numbers is: " + sum);

//2. Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number.

let num = parseInt(prompt("Enter a positive integer:"));

if (num > 0) {
  console.log("Multiplication table of " + num + ":");

  for (let i = 1; i <= 10; i++) {
    let product = num * i;
    console.log(num + " x " + i + " = " + product);
  }
} else {
  console.log("Invalid input, please enter a positive integer.");
}

//3. Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24

let num = parseInt(prompt("Enter a positive integer:"));

if (num >= 0) {
  let factorial = 1;

  for (let i = num; i >= 1; i--) {
    factorial *= i;
  }

  console.log("Factorial of " + num + " = " + factorial);
} else {
  console.log("Invalid input, please enter a non-negative integer.");
}


//4. Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method)

let base = parseInt(prompt("Enter the base number:"));
let exponent = parseInt(prompt("Enter the exponent number:"));

if (exponent >= 0) {
  let result = 1;

  for (let i = 1; i <= exponent; i++) {
    result *= base;
  }

  console.log(base + " raised to the power of " + exponent + " = " + result);
} else {
  console.log("Invalid input, please enter a non-negative integer for exponent.");
}


//5. Determine and print the number of times the character ‘a’ appears in the input entered by the user.

let input = prompt("Enter a string:");

let count = 0;

for (let i = 0; i < input.length; i++) {
  if (input[i] === 'a') {
    count++;
  }
}

console.log("The character 'a' appears " + count + " times in the input.");


//6. Write a program to print a box of #’s taking height and width values from the user.

let height = parseInt(prompt("Enter the height of box :"));
let width = parseInt(prompt("Enter the width of box:"));

      for (let i = 0; i < height; i++) {
        let row = '';
        for (let j = 0; j < width; j++) {
          row += '#';
        }
        console.log(row);
      }


  
  //7. You have to design the code such that the user has only three tries to guess the correct pin of the account. You set the pin as a constant with a final attribute. When correct display “Correct, welcome back.” When incorrect display “Incorrect, try again.”. When ran out of tries display “Sorry but you have been locked out.”

  const PIN = 1234;
  let attempts = 3;
  
  while (attempts > 0) {
    const guess = parseInt(prompt("Enter the 4-digit pin:"));
  
    if (guess === PIN) {
      console.log("Correct, welcome back.");
      break;
    } else {
      attempts--;
      console.log(`Incorrect, try again. You have ${attempts} attempts remaining.`);
    }
  }
  
  if (attempts === 0) {
    console.log("Sorry but you have been locked out.");
  }
  


```

**Comment by chukku10** [comment]
//1. Write a program to calculate the sum of the first 10 natural numbers

let sum = 0;
for (let i = 1; i <= 10; i++) {
  sum += i;
  if (i == 10) {
    console.log("The sum of the first 10 natural numbers is " + sum);
  }
}


//2..Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number. 

let num = prompt("Enter a positive integer:");
if (num > 0) {
  for (let i = 1; i <= 10; i++) {
    console.log(`${num} x ${i} = ${num*i}`);
  }
} else if (num == 0) {
  console.log("The multiplication table of 0 is 0");
} else {
  console.log("Invalid input. Please enter a positive integer.");
}

//3.Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24


let num = prompt("Enter a positive integer:");
let factorial = 1;

if (num >= 0) {
  for (let i = num; i > 0; i--) {
    factorial *= i;
  }
  console.log(`Factorial of ${num} is ${factorial}`);
} else {
  console.log("Invalid input. Please enter a positive integer.");
}


//4..Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method) 

let base = prompt("Enter the base number:");
let exponent = prompt("Enter the exponent number:");
let result = 1;

if (exponent >= 0) {
  for (let i = 1; i <= exponent; i++) {
    result *= base;
  }
  console.log(`${base} raised to the power of ${exponent} is ${result}`);
} else {
  console.log("Invalid input. Exponent should be a non-negative integer.");
}


//5.Determine and print the number of times the character ‘a’ appears in the input entered by the user. 
let input = prompt("Enter a string:");
let count = 0;

for (let i = 0; i < input.length; i++) {
  if (input[i] === "a") {
    count++;
  }
}

console.log(`The character 'a' appears ${count} times in the input.`);

//6.Write a program to print a box of #’s taking height and width values from the user. 

let height = prompt("Enter the height of the box:");
let width = prompt("Enter the width of the box:");

if (height > 0 && width > 0) {
  // Print top edge of the box
  let top = "";
  for (let i = 1; i <= width; i++) {
    top += "#";
  }
  console.log(top);

  // Print middle rows of the box
  for (let i = 2; i <= height - 1; i++) {
    let middle = "#";
    for (let j = 2; j <= width - 1; j++) {
      middle += " ";
    }
    middle += "#";
    console.log(middle);
  }

  // Print bottom edge of the box
  if (height > 1) {
    let bottom = "";
    for (let i = 1; i <= width; i++) {
      bottom += "#";
    }
    console.log(bottom);
  }
} else {
   console.log("Invalid input. Height and width should be positive integers.");
}

7. You have to design the code such that the user has only three tries to guess the correct pin of the account. You set the pin as a constant with a final attribute. When correct display “Correct, welcome back.” When incorrect display “Incorrect, try again.”. When ran out of tries display “Sorry but you have been locked out.”

const PIN = "1234";
let tries = 3;

while (tries > 0) {
  let input = prompt(`Enter the PIN (${tries} tries left):`);
  if (input === PIN) {
    console.log("Correct, welcome back.");
    break;
  } else {
    console.log("Incorrect, try again.");
    tries--;
  }
}

if (tries === 0) {
  console.log("Sorry but you have been locked out.");
}
0 replies
Comment options

sam2elect
now
//1. Write a program to calculate the sum of the first 10 natural numbers.

let sum = 0;
for (let i = 1; i <= 10; i++) {
  sum += i;
}
console.log("The sum of the first 10 natural numbers is: " + sum);

//2. Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number.

let num = parseInt(prompt("Enter a positive integer:"));

if (num > 0) {
  console.log("Multiplication table of " + num + ":");

  for (let i = 1; i <= 10; i++) {
    let product = num * i;
    console.log(num + " x " + i + " = " + product);
  }
} else {
  console.log("Invalid input, please enter a positive integer.");
}

//3. Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24

let num = parseInt(prompt("Enter a positive integer:"));

if (num >= 0) {
  let factorial = 1;

  for (let i = num; i >= 1; i--) {
    factorial *= i;
  }

  console.log("Factorial of " + num + " = " + factorial);
} else {
  console.log("Invalid input, please enter a non-negative integer.");
}


//4. Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method)

let base = parseInt(prompt("Enter the base number:"));
let exponent = parseInt(prompt("Enter the exponent number:"));

if (exponent >= 0) {
  let result = 1;

  for (let i = 1; i <= exponent; i++) {
    result *= base;
  }

  console.log(base + " raised to the power of " + exponent + " = " + result);
} else {
  console.log("Invalid input, please enter a non-negative integer for exponent.");
}


//5. Determine and print the number of times the character ‘a’ appears in the input entered by the user.

let input = prompt("Enter a string:");

let count = 0;

for (let i = 0; i < input.length; i++) {
  if (input[i] === 'a') {
    count++;
  }
}

console.log("The character 'a' appears " + count + " times in the input.");


//6. Write a program to print a box of #’s taking height and width values from the user.

let height = parseInt(prompt("Enter the height of box :"));
let width = parseInt(prompt("Enter the width of box:"));

      for (let i = 0; i < height; i++) {
        let row = '';
        for (let j = 0; j < width; j++) {
          row += '#';
        }
        console.log(row);
      }


  
  //7. You have to design the code such that the user has only three tries to guess the correct pin of the account. You set the pin as a constant with a final attribute. When correct display “Correct, welcome back.” When incorrect display “Incorrect, try again.”. When ran out of tries display “Sorry but you have been locked out.”

  const PIN = 1234;
  let attempts = 3;
  
  while (attempts > 0) {
    const guess = parseInt(prompt("Enter the 4-digit pin:"));
  
    if (guess === PIN) {
      console.log("Correct, welcome back.");
      break;
    } else {
      attempts--;
      console.log(`Incorrect, try again. You have ${attempts} attempts remaining.`);
    }
  }
  
  if (attempts === 0) {
    console.log("Sorry but you have been locked out.");
  }

**Comment by AratiKumari123** [comment]
```
 *  Write a program to calculate the sum of the first 10 natural numbers.

 */
let i=1;
while (i<=10){
    i++;
}
console.log("The sum is " + i );



/**
 * Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number.

 */
let PositiveNumber=prompt("Enter the positive number only");
let i=1;

while (i<=10){
    i*=PositiveNumber;
    console.log(i*PositiveNumber);
    i++;
}

/**
 * Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24

 */
let num=promp("Enter the number ");
let i=1;
while(i>0){
    if(num<0){
        break;
    }
    num=num*i;
}
console.log('the factorial is ${num}');

/**
 * Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method)

 */

let num1=promp("Enter the first number");
let num2=promp("Enter the second number");

while(num2>0){
    num1=num1*num1;
}
console.log('The power of first number to another is ${num1} ');


/**
 * Determine and print the number of times the character ‘a’ appears in the input entered by the user.

 */

let char=prompt("Enter a character");
let count=0;
for(let i=0;i<char.length;i++){
    if(char.charAt(i)=='a'){
        count++;
    }

    /**
     * Write a program to print a box of #’s taking height and width values from the user.

     */
let height=prompt("Enter the Height of the box");
let width=prompt("Enter the width of the box");

for(let i=0;i<height;i++){
    for(let j=0;j<width;j++){
        console.log("#");
    }
}


/**
 * You have to design the code such that the user has only three tries to guess the correct pin of the account. You set the pin as a constant with a final attribute. When correct display “Correct, welcome back.” When incorrect display “Incorrect, try again.”. When ran out of tries display “Sorry but you have been locked out.”

 */
const final =6576;

let count=0;

while(count<3){
    let pin=prompt("Enter the pin");
    if(pin==final){
        console.log("Correct, welcome back.");
        break;
    }
    else {
        console.log("Incorrect, try again.");
    }

}
if(count==4){
    console.log("Sorry but you have been locked out.");
}
```

**Comment by bipin-coded** [comment]
```javascript

// 1. Write a program to calculate the sum of the first 10 natural numbers.
let sum = 0;
let i = 1;

/* Using while loop */
while(i <= 10){
	sum += i;
	i++;
}

/* Using for loop */
for(let j = 1; j <= 10; j++){
	sum += j;
}

console.log('The sum of the first 10 natural numbers is : ' + sum);

// 2..Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number.
let num = +prompt('Enter the positive number');

/* Using while loop */
let i = 1;
while(i<=10){
	console.log(`${num} x ${i} = ${num*i}`);
	i++;
}

/* Using for loop */
for(let i = 1; i <= 10; i++){
	 console.log(`${num} x ${i} = ${num*i}`);
}

// 3. Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24
let num = +prompt('Enter the number');
let factorial = 1;

/* Using while loop */
let i = 1;
while(i <= num){
	factorial *= i;
	i++;
}

/* Using for loop */
for(let i = 1; i <= num; i++) {
  factorial *= i;
}

console.log('Factorial : '+ factorial);

// 4. Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method)
let base = prompt("Enter the base number");
let exponent = prompt("Enter the exponent number");
let result = 1;

/* Using while loop */
let counter = 0;
while(counter <= exponent){
	result *= base;
	counter++;
}

/* Using for loop */
for (let i = 1; i <= exponent; i++) {
	result *= base;
}

console.log(`${base} raised to the power of ${exponent} is ${result}`);

// 5. Determine and print the number of times the character ‘a’ appears in the input entered by the user.
let input = prompt("Enter a string");
let counter = 0;

/* Using while loop */
let i = 0;
while(i < input.length){
	if (input[i] === "a") {
    counter++;
  }
	i++;
}

/* Using foor loop */
for (let i = 0; i < input.length; i++) {
  if (input[i] === "a") {
    counter++;
  }
}

console.log(`The character 'a' appears ${counter} times in the input.`);
```

**Comment by saugatacr7** [comment]
1.// program to display the sum of natural numbers

// take input from the user
const number = parseInt(prompt('Enter a positive integer: '));

let sum = 0;

// looping from i = 1 to number
// in each iteration, i is increased by 1
for (let i = 1; i <= number; i++) {
    sum =i++;
}

console.log('The sum of natural numbers:', sum);

2.// program to generate a multiplication table

// take input from the user
const number = parseInt(prompt('Enter an integer: '));

//creating a multiplication table
for(let i = 1; i <= 10; i++) {

    // multiply i with number
    const result = i * number;

    // display the result
    console.log(`${number} * ${i} = ${result}`);
}

3.// take input from the user
const number = parseInt(prompt('Enter a positive integer: '));

// checking if number is negative
if (number < 0) {
    console.log('Error! Factorial for negative number does not exist.');
}

// if number is 0
else if (number === 0) {
    console.log(`The factorial of ${number} is 1.`);
}

// if number is positive
else {
    let fact = 1;
    for (i = 1; i <= number; i++) {
        fact *= i;
    }
    console.log(`The factorial of ${number} is ${fact}.`);
}

**Comment by vk0027** [comment]
*1 Write a program to calculate the sum of the first 10 natural numbers.
 
let i = 1;
let sum = 0;
while(i<=10){
    sum = sum+i;
    i++;
}
console.log(sum);


*2 Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number.

let num=prompt("enter a positive integer");  //2
let i=1;

while(i<=10){
    console.log(num*i);
    i++;
}

*3 Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24

let num = +prompt('Enter the number');
let factorial = 1;

let i = 1;
while(i <= num){
	factorial *= i;
	i++;
}
console.log('Factorial : '+ factorial);


*4 Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method)

let base = prompt("Enter the base number");
let exponent = prompt("Enter the exponent number");
let result = 1;

let i = 0;
while(i <= exponent){
	result *= base;
	i++;
}
console.log(`${base} raised to the power of ${exponent} is ${result}`);


*5 Determine and print the number of times the character ‘a’ appears in the input entered by the user.

let i=0;
let char = prompt("Enter a string:");

let count = 0;
while(i < char.length) {
if (char[i] === 'a') {
count++;
}
}
console.log(count);


*6 write a program to print a box of #’s taking height and width values from the user.

let height = prompt("Enter the height of the box:");
let width = prompt("Enter the width of the box:");

if (height > 0 && width > 0) {
  let top = "";
  for (let i = 1; i <= width; i++) {
    top += "#";
  }
  console.log(top);

  for (let i = 2; i <= height - 1; i++) {
    let middle = "#";
    for (let j = 2; j <= width - 1; j++) {
      middle += " ";
    }
    middle += "#";
    console.log(middle);
  }

  if (height > 1) {
    let bottom = "";
    for (let i = 1; i <= width; i++) {
      bottom += "#";
    }
    console.log(bottom);
  }
} else {
   console.log("Invalid input. Height and width should be positive integers.");
}


*7 You have to design the code such that the user has only three tries to guess the correct pin of the account. You set the pin as a constant with a final attribute. When correct display “Correct, welcome back.” When incorrect display “Incorrect, try again.”. When ran out of tries display “Sorry but you have been locked out.”

let pin=246;
let attempts=3;
while (attempts>0){
    let guess=parseInt(prompt("enter 3-digit pin :"));
    if(guess===pin){
        console.log('correct pin!');
        break;
    }
    else{
        attempts--;
        console.log("incorrect,try again!");
    }
}
if(attempts===0){
    console.log("incorrect, you are locked out!")

**Comment by kumarneeraj89** [comment]
# Code

```javascript
//Write a program to calculate the sum of the first 10 natural numbers.

let counter = 1;
let sum = 0;
while(counter<=10){
    sum = sum+counter;
    counter++;
}
console.log(sum);

//Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number.


let positive_integer = +prompt('Enter a positive integer: ');
let counter = 1;
table=1;

if(positive_integer>0){
    while(counter<=10){
        table = counter*positive_integer;
        console.log(`${positive_integer} x ${counter} = ${table}`);
        counter++;
    }
}
else{
    console.log('input is not a positive integer');
}

//Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24

let n = +prompt('Enter a number: ');
let output = "";
console.log(`Factors of Given Number: ${n} as below`)
/*while(counter<=n){
    y = n%counter;
    if (y===0){
        output = output + counter + ",";
    }
    counter = counter+1;
}
*/
for(let counter = 1; counter<=n; counter++){
    y = n%counter;
    if (y===0){
        output = output + counter + ",";
    }
}
console.log("{ " + output + "}");

//Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method)

let n = +prompt('Enter a number: ');
let m = +prompt('Enter a power number: ');
let z = n;
let y = 1;
counter = 1;
while(counter<=m){
     y = y*n;
    counter = counter+1;
}
console.log(`Value of ${z} raised to the power of ${m} is: ${y}`)

//Determine and print the number of times the character ‘a’ appears in the input entered by the user.
let n = prompt('Enter a string: ');
let m = n.length;
let chars_s=0;
counter=0;
while(counter<m) {
    if(n.charAt(counter)=='a') {
        chars_s = chars_s+1;
    }
    counter = counter+1;
}
console.log(`The number of times the character ‘a’ appears in the ${n} entered by the user is: ${chars_s}.`);

//Write a program to print a box of #’s taking height and width values from the user..
let n = prompt('Enter the height of the box: ');
let m = prompt('Enter the width of the box: ');

for(i=1;i<=n;i++){
    let output="";
    for(j=1;j<=m;j++){
        output = output + '#';
    }
    console.log(output);
}

```

**Comment by Ravi-710** [comment]
// 1. Write a program to calculate the sum of the first 10 natural numbers.
let sum = 0;
for (let i = 1; i <= 10; i++)
 {
  sum += i;
 }
console.log("The sum of the first 10 natural numbers is " + sum);

// 2..Write a program that prompts the user to input a positive integer. It should then print the multiplication table of that number.
const prompt=require("prompt-sync")({sigint:true}); 
let number = prompt("enter value");
if (number > 0) {
  console.log("Multiplication table of " + number + ":");
  for (let i = 1; i <= 10; i++) {
    let product = number * i;
    console.log(number + " x " + i + " = " + product);
  }
}
 else 
 {
  console.log("Invalid input, please enter a positive integer.");
}

//3 Write a program to find the factorial value of any number entered through the keyboard. Example: factorial 4 = 4 * 3 * 2 * 1 = 24
const prompt=require("prompt-sync")({sigint:true});
let number = prompt("Enter a positive integer:");
let factorial = 1;

if (number >= 0) {
for (let i = number; i > 0; i--) {
factorial *= i;
}
console.log("Factorial of "+number +"is" + factorial);
}
 else {
console.log("Invalid input. Please enter a positive integer.");
}
//4..Two numbers are entered through the keyboard. Write a program to find the value of one number raised to the power of another. (Do not use JavaScript built-in method)
const prompt=require("prompt-sync")({sigint:true});
let base = prompt("Enter the base number:");
let exponent = prompt("Enter the exponent number:");
let result = 1;

if (exponent >= 0) 
{
  for (let i = 1; i <= exponent; i++) {
    result *= base;
  }
  console.log(`${base} raised to the power of ${exponent} is ${result}`);
} else 
{
  console.log("Invalid input. Exponent should be a non-negative integer.");
}
//5 Determine and print the number of times the character ‘a’ appears in the input entered by the user.
const prompt=require("prompt-sync")({sigint:true});
let input = prompt("Enter a string:");
let count = 0;
for (let i = 0; i < input.length; i++) {
if (input[i] === 'a') {
count++;
}
}
//6. Write a program to print a box of #’s taking height and width values from the user.
const prompt=require("prompt-sync")({sigint:true});
let height = parseInt(prompt("Enter the height of box :"));
let width = parseInt(prompt("Enter the width of box:"));
  for (let i = 0; i < height; i++) {
    let row = '';
    for (let j = 0; j < width; j++) {
      row += '#';
    }
    console.log(row);
  }
  //7. You have to design the code such that the user has only three tries to guess the correct pin of the account. You set the pin as a constant with a final attribute. When correct display “Correct, welcome back.” When incorrect display “Incorrect, try again.”. When ran out of tries display “Sorry but you have been locked out.”
  const prompt=require("prompt-sync")({sigint:true});
  const PIN = 1234;
  let attempts = 3;
  
  while (attempts > 0) {
    const guess = parseInt(prompt("Enter the 4-digit pin:"));
  
    if (guess === PIN) {
      console.log("Correct, welcome back.");
      break;
    } else {
      attempts--;
      console.log(`Incorrect, try again. You have ${attempts} attempts remaining.`);
    }
  }
  
  if (attempts === 0) {
    console.log("Sorry but you have been locked out.");
  }

**Comment by RevathiSundaramoorthy** [comment]
## Code
```javascript
//sum of first 10 natural numbers
let sum = 0
for (let counter1=1; counter1<=10; counter1++) {
sum += counter1;
}
console.log("Sum of first 10 natural numbers: " + sum);
//print multiplication table of a number
let num1 = parseInt(prompt("Enter a number to print multiplication table"));
const num2 = 10;
if (num1 > 0) {
    console.log("Printing multiplication table of " + num1);
    for (let counter=1; counter<=num2; counter++) {
        console.log (counter + " * " + num1 + " = " + num1*counter);
    }
}
else {
    console.log ("Input is not a positive number, cannot print multiplication table");
}
//factorial value
let numInput = parseInt(prompt("Enter a number to find factorial"));
if (numInput > 0) {
    let numFactorial = 1;
    let x = numInput
    for (; x>=1; x--) {
        numFactorial = numFactorial * x;
    }
    console.log ("Factorial of " + numInput + " is: " + numFactorial);
}
else {
    console.log ("Enter a positive integer");
}
//value of one number raised to the power of another
let num_base = parseInt(prompt("Enter a number for base component"));
let num_expo = parseInt(prompt("Enter a number for exponent component"));
if ((num_base > 0) && (num_expo > 0)) {
    let result = 1;
    for (y=1; y<=num_expo; y++) {
        result = result * num_base
    }
    console.log (num_base + " raised to the power of " + num_expo + " is: " + result);

}
else {
    console.log ("Enter positive integers");
}
//check for occurence of 'a'
let str = prompt("Enter the input string to check for 'a' ");
let str_count = 0;
for (z=0; z<str.length; z++) {
    if (str[z] == 'a') {
        str_count++;
    }
}
console.log ("Input string: " + str);
console.log ("Number of times 'a' occured in the input string: " + str_count);
//print box of #
let height = prompt("Enter the height of the box");
let width = prompt("Enter the width of the box");
console.log ("Height entered: " + height);
console.log ("Width entered: " + width);
if ((height > 0) && (width > 0)) {
    let str_row1 = '#';
    let str_row2 = '#';
    for (let a=2; a<=width; a++) {
        str_row1 = str_row1 + '#';
        if (a==width) {
        str_row2 = str_row2 + '#';
        }
        else {
        str_row2 = str_row2 + ' ';
        }
    }
    for (let b=1; b<=height; b++) {
        if ((b==1) || (b==height)) {
        console.log (str_row1);
        }
        else {
        console.log (str_row2);
        }
    }
}
else {
    console.log("Please enter values greater than zero");
}
//verify pin
const pin = 9999;
let correctPin = false;
  let inputPin = parseInt(prompt("Enter the 4-digit pin"));
for (p=1; ((p<=3) && (correctPin == false)); p++) {
  
    if (inputPin === pin) {
        correctPin = true;
        console.log ("Correct, welcome back.");
    }
    else {
        if (p == 3) {
		    console.log ("Sorry but you have been locked out");        
        }
        else {
            console.log ("Incorrect, try again.");
            inputPin = parseInt(prompt("Incorrect, try again. Enter the 4-digit pin"));
        }
        
    }
}
```
## Screenshot
![image](https://user-images.githubusercontent.com/126975777/231574498-53e9160a-20bf-4513-9303-e1cc7e619307.png)
![image](https://user-images.githubusercontent.com/126975777/231574654-2c8ba840-e7a0-4647-b40e-f7fe83333581.png)
![image](https://user-images.githubusercontent.com/126975777/231574837-97ce0058-1975-4420-ab00-59ec1b2a3cb1.png)
![image](https://user-images.githubusercontent.com/126975777/231575600-e7818b42-6265-4c3a-950e-46ddc8ea792a.png)
